### ISSUE TITLE
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException at `CleverTapAPI#showInApp` v3.7.2 

### ISSUE THREAD

**Issue by wisnukurniawan** [CleverTap/clevertap-android-sdk#94]
Hi Clevertap team, we still got the same error with this issue https://github.com/CleverTap/clevertap-android-sdk/issues/69 after updating to 3.7.2

```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=10; index=-1
       at java.util.ArrayList.add(ArrayList.java:468)
       at com.clevertap.android.sdk.CleverTapAPI.showInApp(CleverTapAPI.java:664)
       at com.clevertap.android.sdk.CleverTapAPI.access$800(CleverTapAPI.java:97)
       at com.clevertap.android.sdk.CleverTapAPI$6.run(CleverTapAPI.java:649)
       at android.os.Handler.handleCallback(Handler.java:883)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:237)
       at android.app.ActivityThread.main(ActivityThread.java:7811)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1068)
```
<img width="441" alt="Screen Shot 2020-08-19 at 11 03 27" src="https://user-images.githubusercontent.com/16729158/90590846-bf496f80-e20b-11ea-8ef4-92a9c39bcda0.png">
<img width="440" alt="Screen Shot 2020-08-19 at 11 03 34" src="https://user-images.githubusercontent.com/16729158/90590850-c2446000-e20b-11ea-9902-a6d01c5aa1f1.png">
<img width="440" alt="Screen Shot 2020-08-19 at 11 03 43" src="https://user-images.githubusercontent.com/16729158/90590852-c40e2380-e20b-11ea-8440-c08fc81ff119.png">
<img width="435" alt="Screen Shot 2020-08-19 at 14 27 38" src="https://user-images.githubusercontent.com/16729158/90605251-4573af00-e228-11ea-93b4-6fd163da8736.png">
<img width="441" alt="Screen Shot 2020-08-19 at 11 04 02" src="https://user-images.githubusercontent.com/16729158/90590856-c6707d80-e20b-11ea-9cc7-fc91cbafc1aa.png">

We have put crashlytics log in our activity lifecycle, the crash happens after user close the app. It turns out by last crash record is `HomeActivity.onPause`

<img width="1061" alt="Screen Shot 2020-08-19 at 11 06 59" src="https://user-images.githubusercontent.com/16729158/90591088-60382a80-e20c-11ea-809e-e9fe432b0f26.png">

After doing research, that crash is actually happening because of accessing ArrayList from a different thread, please take a look https://developer.android.com/reference/java/util/ArrayList

```
Note that this implementation is not synchronized. If multiple threads access an ArrayList instance concurrently, and at least
one of the threads modifies the list structurally, it must be synchronized externally. (A structural modification is any operation
that adds or deletes one or more elements, or explicitly resizes the backing array; merely setting the value of an element is
not a structural modification.) This is typically accomplished by synchronizing on some object that naturally encapsulates the
list. If no such object exists, the list should be "wrapped" using the Collections#synchronizedList method. This is best done at
creation time, to prevent accidental unsynchronized access to the list:
```

**Comment by piyush-kukadiya** [comment]
Thanks @wisnukurniawan for the detailed info. We will look into this and notify you once this fix is released.

**Comment by darshanclevertap** [comment]
@wisnukurniawan We have fixed this in the CleverTap Android SDK v4.0.0. This is a major release so please check out the [changes here](https://github.com/CleverTap/clevertap-android-sdk/blob/master/docs/CTV4CHANGES.md) which we have made before updating.

**Comment by wisnukurniawan** [comment]
Ack will check that. thank you!
